code/modules/mob/living/basic/guardian/guardian_verbs.dm

```diff
@@ -0,0 +1,185 @@
+	background_icon_state = "base"
+	click_to_activate = FALSE
+	cooldown_time = 0 SECONDS
+	melee_cooldown_time = 0
+	shared_cooldown = NONE
+
+/datum/action/cooldown/mob_cooldown/recall_guardian/Activate(atom/target)
+	var/mob/living/living_owner = owner
+	var/list/guardians = living_owner.get_all_linked_holoparasites()
+	for(var/mob/living/basic/guardian/guardian in guardians)
+		guardian.recall()
+	StartCooldown()
+	return TRUE
+
+/// Replace an annoying griefer you were paired up to with a different but probably no less annoying player.
+/datum/action/cooldown/mob_cooldown/replace_guardian
+	name = "Reset Guardian Consciousness"
+	desc = "Replaces the mind of your guardian with that of a different ghost."
+	button_icon = 'icons/mob/simple/mob.dmi'
+	button_icon_state = "ghost"
+	background_icon = 'icons/hud/guardian.dmi'
+	background_icon_state = "base"
+	click_to_activate = FALSE
+	cooldown_time = 5 SECONDS
+	melee_cooldown_time = 0
+	shared_cooldown = NONE
+
+/datum/action/cooldown/mob_cooldown/replace_guardian/Activate(atom/target)
+	StartCooldown(5 MINUTES)
+
+	var/mob/living/living_owner = owner
+	var/list/guardians = living_owner.get_all_linked_holoparasites()
+	for(var/mob/living/basic/guardian/resetting_guardian as anything in guardians)
+		if (!COOLDOWN_FINISHED(resetting_guardian, resetting_cooldown))
+			guardians -= resetting_guardian //clear out guardians that are already reset
+
+	if (!length(guardians))
+		to_chat(owner, span_holoparasite("You cannot reset [length(guardians) > 1 ? "any of your guardians":"your guardian"] yet."))
+		StartCooldown()
+		return FALSE
+
+	var/mob/living/basic/guardian/chosen_guardian = tgui_input_list(owner, "Pick the guardian you wish to reset", "Guardian Reset", sort_names(guardians))
+	if (isnull(chosen_guardian))
+		to_chat(owner, span_holoparasite("You decide not to reset [length(guardians) > 1 ? "any of your guardians":"your guardian"]."))
+		StartCooldown()
+		return FALSE
+
+	to_chat(owner, span_holoparasite("You attempt to reset <font color=\"[chosen_guardian.guardian_colour]\">[span_bold(chosen_guardian.real_name)]</font>'s personality..."))
+	var/list/mob/dead/observer/ghost_candidates = poll_ghost_candidates("Do you want to play as [owner.real_name]'s [chosen_guardian.theme.name]?", ROLE_PAI, FALSE, 100)
+	if (!LAZYLEN(ghost_candidates))
+		to_chat(owner, span_holoparasite("There were no ghosts willing to take control of <font color=\"[chosen_guardian.guardian_colour]\">[span_bold(chosen_guardian.real_name)]</font>. Looks like you're stuck with it for now."))
+		StartCooldown()
+		return FALSE
+
+	var/mob/dead/observer/candidate = pick(ghost_candidates)
+	to_chat(chosen_guardian, span_holoparasite("Your user reset you, and your body was taken over by a ghost. Looks like they weren't happy with your performance."))
+	to_chat(owner, span_boldholoparasite("Your <font color=\"[chosen_guardian.guardian_colour]\">[chosen_guardian.theme.name]</font> has been successfully reset."))
```

Comment:
```suggestion
	to_chat(owner, span_boldholoparasite("The personality of <font color=\"[chosen_guardian.guardian_colour]\">[chosen_guardian.theme.name]</font> has been successfully reset."))
```

just to make it clear you rerolled the player instead of maybe rerolling the type of holoparasite or whatever